### ISSUE TITLE
config.SUDO not respected for 2.0.1

### ISSUE THREAD

**Issue by jmpolom** [Fizzadar/pyinfra#794]
## Describe the bug
A deploy beginning with:
```
from pyinfra import config
from pyinfra.operations import files, server, systemd

config.SUDO = True

(etc)
```

does not seem to result in every fact or operation being executed with sudo. This worked in 1.x as recently as 1.7.3 (as expected? this is really not documented).

Perhaps I am dense and missing some necessary magical incantations to get this working in 2.x. The changelog notes "config variables must now be set on the global config object" for 2.0.0 but a specific example isn't given.

It seems like setting `_sudo` to `True` in the host variables does result in every operation/fact executing with sudo however.

## To Reproduce
Above shows reasonable context that should result in reproducible behavior.

## Expected behavior
`config.SUDO = True` in a deploy should result in every operation/fact executing with sudo on the target host.

## Meta
+ Include output of `pyinfra --support`.
```
--> Support information:
    System: Darwin
      Platform: macOS-12.3.1-arm64-arm-64bit
      Release: 21.4.0
      Machine: arm64
    pyinfra: v2.0.1
    Executable: /Users/jon/.local/bin/pyinfra
    Python: 3.10.2 (CPython, Clang 13.0.0 (clang-1300.0.29.3))
```
+ How was pyinfra installed (source/pip)? pipx and also pip3 in a venv
+ Include pyinfra-debug.log (if one was created)
+ Consider including output with `-vv` and `--debug`.

**Comment by jmpolom** [comment]
Looks like this was fixed in commit 88bbbeeb0ca0e2cda0690c532207b874205974ff -- probably worth making a new release as this seems like a pretty major bug breaking an essential feature (priv esc).

**Comment by Fizzadar** [comment]
Agreed; I have just released a hotfix `v2.0.2` that includes the commit!
